src/qibolab/result.py

```diff
@@ -1,100 +1,91 @@
-        if data.shots is not None:
-            shots = np.append(self.shots, data.shots, axis=axis)
-        else:
-            shots = None
-
-        new_execution_results = self.__class__.from_components(i, q, shots)
-
-        return new_execution_results
+        return self.voltage.q
 
     @cached_property
-    def measurement(self):
-        """Resonator signal voltage mesurement (MSR) in volts."""
+    def magnitude(self):
+        """Signal magnitude in volts."""
         return np.sqrt(self.i**2 + self.q**2)
 
     @cached_property
     def phase(self):
-        """Computes phase value."""
+        """Signal phase in radiants."""
         phase = np.angle(self.i + 1.0j * self.q)
         return phase
-        # return signal.detrend(np.unwrap(phase))
 
     @cached_property
-    def ground_state_probability(self):
-        """Computes ground state probability"""
-        return 1 - np.mean(self.shots)
-
-    def raw_probability(self, state=1):
-        """Serialize probabilities in dict.
-        Args:
-            state (int): if 0 stores the probabilities of finding
-                        the ground state. If 1 stores the
-                        probabilities of finding the excited state.
-        """
-        if state == 1:
-            return {"probability": 1 - self.ground_state_probability}
-        elif state == 0:
-            return {"probability": self.ground_state_probability}
+    def state_0_probability(self):
+        return self.probability(0) if self.state else None
 
-    @property
-    def average(self):
-        """Perform average over i and q"""
-        return AveragedResults.from_components(np.mean(self.i), np.mean(self.q))
+    @cached_property
+    def state_1_probability(self):
+        return self.probability(1) if self.state else None
 
-    @property
-    def raw(self):
-        """Serialize output in dict."""
+    def probability(self, state=0):
+        """Returns the statistical frequency of the specified state (0 or 1)."""
+        return np.count_nonzero(self.state == state) / len(self.state)
 
-        return {
-            "MSR[V]": self.measurement,
+    def __add__(self, data):  # __add__(self, data:ExecutionResults) -> ExecutionResults
+        axis = 0 if len(data.i.shape) > 0 else None
+        i = np.append(self.i, data.i, axis=axis)
+        q = np.append(self.q, data.q, axis=axis)
+        if data.state:
+            state = np.append(self.state, data.state, axis=axis)
+        else:
+            state = None
+        return ExecutionResults(i, q, state)
+
+    def __len__(self):
+        return len(self.i)
+
+    def serialize(self):
+        """Serialize as a dictionary."""
+        serialized_dict = {
+            "magnitude[V]": self.magnitude,
             "i[V]": self.i,
             "q[V]": self.q,
             "phase[rad]": self.phase,
         }
+        if self.state:
+            serialized_dict["state"] = self.state_1_probability
+        return serialized_dict
```

Comment:
A method with just `self`, returning an object, it is well represented by the Python `property` abstraction.

Reply:
Hi Alessandro,
I disagree with you here.
It is as natural for an object to have properties as it is to have methods.
A lamp object may have different properties like `color` or `brand`, and methods like `turn_on()` and `turn_off()`.
Just because turn_on() does not have any arguments, you would not convert it into a property, would you?.
In our specific case, the fact that we could not find a right name for the property, already tells that it doesn't fit well as a property. On the other hand we all understand what it means to serialize and object.
I think, at least in this case, it is important to keep the method, for the sake of making code that reads well and is understood well.